### PR TITLE
Round 1: A Jasmine Workshop

### PR DESCRIPTION
**Pull request by brendajin**

### CODE REVIEW

**Comment by brendajin** [comment]
@pamelafox Let me know what you think!

File: jasmine/index.html [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+						<li>break down the pieces</li>
+						<li>ensure ease of upgrades and deprecations</li>
+					</ul>
+				</section>
+				<section>
+					<h3>Benefits of testing</h3>
+					<ul>
+						<li>shorten feedback loop</li>
+						<li>prevent regressions</li>
+						<li>document code</li>
```
Comment (pamelafox) [inline jasmine/index.html]: You could use the speakers not feature to explain these slides in more detail, how you'd talk over these points.

File: jasmine/index.html [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+				</section>
+				<section>
+					<h3>What is TDD?</h3>
+					<br>
+					<p><strong>Test-driven Development (TDD)</strong> is an approach to programming.</p>
+					<br>
+					<ol>
+						<li>Write an automated (failing) test. This test will describe the desired behavior
+						<li>Write a small amount of code that will make the test pass</li>
+						<li>Repeat</li>
```
Comment (pamelafox) [inline jasmine/index.html]: Do you think it'd help to compare that to the style of programming where we write code first and then write tests? not sure if theres a name for that.

File: jasmine/index.html [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+				</section>
+				<section>
+					<h3>What is BDD?</h3>
+					<br>
+					<p><strong>Behavior-driven Development (BDD)</strong> emerged out of TDD as an approach to programming that considers both business and technical needs.</p>
+					<br>
+					<p>From the developer perspective, BDD is a style of TDD that uses English in a natural way to describe expected behavior.</p>
```
Comment (pamelafox) [inline jasmine/index.html]: Its taken me a while to get BDD versus non BDD. Can you contrast here?

File: jasmine/index.html [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+						<pre><code>Describe "a part of an application"
+	It "should do something specific"
+</code></pre>
+					</p>
+					<br>
+					<p>So you might have something like this:
+						<pre><code>Describe "a cat trampoline"
+	It "should have a place for a cat to jump"
+	It "should not be large enough for a person"
+	It "should make cats happy"
```
Comment (pamelafox) [inline jasmine/index.html]: Ohhh now I get BDD.
Okay maybe this slide is enough to explain it.

File: jasmine/index.html [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+					<p>Suggested time: 30 minutes</p>
+				</section>
+
+				<!-- Jasmine Part 2 -->
+				<section>
+					<h3>Control Flow</h3>
+					<h4>Shoulda, coulda, woulda</h4>
+					<p><strong>Control Flow</strong> is the order in which the individual statements, instructions or function calls are executed or evaluated.</p>
+					<br>
+					<p>Because this can get complicated quickly, we'll need to test this.</p>
```
Comment (pamelafox) [inline jasmine/index.html]: Hm. Not sure I understand the distinction between testing control flow and testing other aspects, can you clarify? This slide feels a bit ambiguous to me. Maybe an example?

File: jasmine/index.html [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+    // use matchers to see if the function was called
+    expect(robot.lookLeft).toHaveBeenCalled();
+    expect(robot.lookLeft.calls.count()).toBeGreaterThan(1);
+  });
+});</code></pre>
+				</section>
+				<section>
+					<h3>Tracking Spies</h3>
+					<br>
+					<p>function calls to spies are tracked through the <code>calls</code> property.</p>
```
Comment (pamelafox) [inline jasmine/index.html]: capitalize F

File: jasmine/index.html [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+					<pre><code>expect(robot.lookLeft.calls.count()).toBeGreaterThan(1);</code></pre>
+					<br>
+					<p><a href="http://jasmine.github.io/2.0/introduction.html#section-Other_tracking_properties" target="_blank">Read more about it here!</a></p>
+				</section>
+				<section>
+					<h3>Exercise time</h3>
+					<br>
+					<p><a href="step2/instructions.html" target="_blank">Click here for directions</a></p>
+					<p>Suggested time: 30 minutes</p>
```
Comment (pamelafox) [inline jasmine/index.html]: I'm guessing the exercises will take more like 45, but you can monitor and modify after.

File: jasmine/index.html [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+				</section>
+
+				<!-- End -->
+				<section>
+					<h2>More information</h2>
+					<ul>
+						<li>Testing is the approach, not the goal</li>
+						<li>Some things are hard to test, so get creative</li>
+						<li>Not all tests run in the DOM</li>
```
Comment (pamelafox) [inline jasmine/index.html]: The third point might benefit from clarification. Maybe add a slide that talks about other testing libraries/approaches for them to explore?
Like integration testing, faked dom testing, visual diff testing, etc.

File: jasmine/index.html (hunk 2 above)
Comment (brendajin) [inline jasmine/index.html]: Good call!

File: jasmine/index.html (hunk 1 above)
Comment (brendajin) [inline jasmine/index.html]: Added some more speaker notes below

**Comment by brendajin** [comment]
@pamelafox I made some edits over the weekend. LMK what you think!

**Comment by pamelafox** [comment]
LGTM, merge at will!

**Pull request merged**